src/PKSim.Core/Snapshots/Mappers/SnapshotMapper.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Utility.Collections;
+
+namespace PKSim.Core.Snapshots.Mappers
+{
+   public interface ISnapshotMapper
+   {
+      /// <summary>
+      ///    Given a <paramref name="model" /> object, returns the corresponding snapshot.
+      /// </summary>
+      /// <exception cref="SnapshotNotFoundException">
+      ///    is thrown if a snapshot could not be found for the given
+      ///    <paramref name="model" />
+      /// </exception>
+      object MapToSnapshot(object model);
+
+      /// <summary>
+      ///    Given a <paramref name="snapshot" /> object, returns the corresponding model.
+      /// </summary>
+      /// <exception cref="SnapshotNotFoundException">
+      ///    is thrown if a snapshot could not be found for the given
+      ///    <paramref name="snapshot" />
+      /// </exception>
+      object MapToModel(object snapshot);
+   }
+
```

Comment:
Why `object MapTo`? Shouldn't we define it like
```
public interface ISnapshotMapper<TModel, TSnapshot>
{
    TSnapshot MapToSnapshot(TModel model);
    TModel MapToModel(TSnapshot snapshot);
}
```
?

Reply:
This would not help. I need a interface that is common to all mappers so that it can be called in a very simple way.  This can be seen in the SnapshotMapper implementation. It finds the matching mapper for a given model type

Reply:
I should add that the base implementation has the generic method that you described. The interface  not however. It would be too specific otherwise